src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -802,6 +802,7 @@ private BoundLambda ReallyBind(NamedTypeSymbol delegateType, bool inExpressionTr
             ParameterHelpers.EnsureNativeIntegerAttributeExists(compilation, lambdaParameters, diagnostics, modifyCompilation: false);
             ParameterHelpers.EnsureScopedRefAttributeExists(compilation, lambdaParameters, diagnostics, modifyCompilation: false);
             ParameterHelpers.EnsureNullableAttributeExists(compilation, lambdaSymbol, lambdaParameters, diagnostics, modifyCompilation: false);
+            ParameterHelpers.EnsureRequiresLocationAttributeExists(compilation, lambdaParameters, diagnostics, modifyCompilation: false, moduleBuilder: null);
```

Comment:
>EnsureRequiresLocationAttributeExists [](http://example.com/codeflow?start=29&length=37)

Either ```EnsureRequiresLocationAttributeExists``` and ```EnsureIsReadOnlyAttributeExists``` should be merged, or ```EnsureRequiresLocationAttributeExists``` calls should be placed next to ```EnsureIsReadOnlyAttributeExists``` calls. They are closely related and doing one without another is a mistake.